services/cloudapi/cloudapinic/nic.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			log.Printf("[DEBUG] no nic found for datacenter %s, server %s", datacenterId, serverId)
+		}
+		return nil, apiResponse, err
+	}
+	return &nic, apiResponse, nil
+}
+
+func (fs *Service) Delete(ctx context.Context, datacenterID, serverID, ID string) (*ionoscloud.APIResponse, error) {
+	apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsDelete(ctx, datacenterID, serverID, ID).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return apiResponse, err
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutDelete).WaitForStateContext(ctx)
+	if errState != nil {
+		return apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, ID: %s, Response: (%w)", datacenterID, serverID, ID, errState)
+	}
+	return apiResponse, nil
+}
+
+// Create - creates the resource in the backend and waits until it is in an `AVAILABLE` state
+func (fs *Service) Create(ctx context.Context, datacenterID, serverID string, nic ionoscloud.Nic) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	val, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsPost(ctx, datacenterID, serverID).Nic(nic).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("an error occured while creating val rule for dcId: %s, server_id: %s, Response: (%w)", datacenterID, serverID, err)
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
+	if errState != nil {
+		if cloudapi.IsRequestFailed(err) {
+			// Request failed, so resource was not created, delete resource from state file
+			fs.D.SetId("")
+		}
+		return nil, apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, Response: (%w)", datacenterID, serverID, errState)
+	}
+	return &val, apiResponse, nil
+}
+
+func (fs *Service) Update(ctx context.Context, datacenterID, serverID, ID string, nicProperties ionoscloud.NicProperties) (*ionoscloud.Nic, *ionoscloud.APIResponse, error) {
+	updatedNic, apiResponse, err := fs.Client.NetworkInterfacesApi.DatacentersServersNicsPatch(ctx, datacenterID, serverID, ID).Nic(nicProperties).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return nil, apiResponse, fmt.Errorf("an error occured while updating nic for dcId: %s, server_id: %s, id %s, Response: (%w)", datacenterID, serverID, ID, err)
+	}
+	// Wait, catching any errors
+	_, errState := cloudapi.GetStateChangeConf(fs.Meta, fs.D, apiResponse.Header.Get("Location"), schema.TimeoutUpdate).WaitForStateContext(ctx)
+	if errState != nil {
+		return nil, apiResponse, fmt.Errorf("an error occured while waiting for state change dcId: %s, server_id: %s, ID: %s, Response: (%w)", datacenterID, serverID, ID, errState)
```

Comment:
@cristiGuranIonos same thing as above, please double check all error messages and make sure they are clear.